doc/rst/developer/bestPractices/DCO.rst

```diff
@@ -0,0 +1,131 @@
+git commits must be signed by their developers to certify that they
+wrote, or otherwise have the right to submit, the changes to the
+project.  The full text of the DCO is available in Chapel's
+`CONTRIBUTING.md`_ file.
+
+This means that the message for each commit you submit must contain a
+line that reads:
+
+    Signed-off-by: Random J Developer <random@developer.example.org>
+
+where the name and email address must match those of your GitHub
+account as indicated by `git config user.name` and `git config
+user.email`.  The presence of this line signifies that you're making
+the commit in accordance with the DCO.
+
+The rest of this document describes some tips and tricks for signing
+commits proactively or retroactively.
+
+.. _CONTRIBUTING.md: https://github.com/chapel-lang/chapel/blob/master/.github/CONTRIBUTING.md
+
+
+Ways to Sign Commits
+--------------------
+
+#. Signing each commit manually
+
+Of course, when you are making a commit and editing the commit
+message, you can sign it manually, by typing the line above but with
+your GitHub account's name and email address.  However, this has the
+obvious downsides of being tedious and error-prone.
+
+
+#. Using `git -s`
+
+Git has a `-s | --signoff` command-line flag that will automatically
+add your 'Signed-off-by' line to your commit message.  For an
+interactive commit like the following:
+
+    git commit -s
+
+you should see the 'Signed-off-by' line in your editor when it brings
+up the buffer representing the commit message.  If you supply the
+commit message on the command-line, the 'Signed-off-by' line will be
+automatically added for you.
+
+    git commit -s -m "This is my commit message"
+
+This approach has the advantage of being fairly straightforward and
+requiring no configuration, but the downside that you need to remember
+to use the flag with each commit.
+
+
+#. Using a git commit template
+
+This approach uses a template file that contains your signature for
+interactive commit messages.  It has the advantage of being fairly
+straightforward and making your signature visible to you when you
+commit.  However, a downside is that it doesn't apply your signature
+to git commits that skip the interactive editing of commit messages,
+like `git revert` or `git commit -m ...`.
+
+To take this approach, create a file (say, `~/.gitmessage`) that will
+form the basis for any interactive commit messages, containing your
+signature line:
+
+.. code-block:: bash
+
+     ---
+     Signed-off-by: Random J Developer <random@developer.example.org>
+
+Then, tell your git configuration to open this file for any new
+interactive commit messages using one of the following forms.  To set
+this up for the current repository only:
+
+.. code-block:: bash
+
+     git config commit.template ~/.gitmessage
+
+To set it up across repositories via your `~/.gitconfig` file:
+     
+.. code-block:: bash
+
+     git config --global commit.template ~/.gitmessage
+     
+
+#. Using a git commit hook
```

Comment:
@ronawho: Since you're using the git hook technique, would you do a quick sanity check that I captured the approach correctly?  (i.e., that the instructions are correct and that I've summarized the tradeoffs appropriately / fairly?)  I ended up putting this approach last because, in a sense, the document builds from the most naive to the most bullet-proof approach (however, I'm still using the previous approach because I like seeing the commit line better and haven't been bitten by the downsides... yet).

Reply:
Personally, I don't view having to set hooks up for each repository as a downside. I don't want to sign off for commits to other repos like arkouda, internal repos, or 3rd party libraries that don't require DCO. I might just label that as a difference rather than listing it as an advantage or disadvantage.

Might be nice to add a copy paste command  of `cp $CHPL_HOME/util/misc/pre-commit-msg-hook $CHPL_HOME/.git/hooks/commit-msg` instead of having that in sentence form?

Otherwise looks good to me. 

---

BTW I do see the signoff in my commit messages, which made me wonder what I'm doing differently. I thought I was using something like the template in our repo, but it turns out I do something different that results in populating my template when I'm doing a commit (though it still auto-adds on rebase/amend/whatever.) 

I have a `.git/hooks/prepare-commit-msg` with:

```sh
#!/bin/sh

NAME=$(git config user.name)
EMAIL=$(git config user.email)

if [ -z "$NAME" ]; then
    echo "empty git config user.name"
    exit 1
fi

if [ -z "$EMAIL" ]; then
    echo "empty git config user.email"
    exit 1
fi

git interpret-trailers --if-exists doNothing --trailer \
    "Signed-off-by: $NAME <$EMAIL>" \
    --in-place "$1"
```

But that requires git 2.15 or newer, so it's not as portable. I use something else on internal systems with an older git. I'll try switching over to just that and see how portable it ends up being. If that works well I'll open up a followup PR for the hook docs.

Reply:
Huh... I'd seen `prepare-commit-msg` but had brushed it off for a reason that I can't recall offhand.  Maybe I was assuming that it wouldn't work for the same forms as the `commit.template` doesn't... that it essentially prepared the buffer when a buffer was used.